src/assets/htmx/htmx.md

```diff
@@ -0,0 +1,374 @@
+  <div>item 1</div>
+</div>
+```
+
+After adding another item (item 2)
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+  <div>item 2</div>
+</div>
+```
+
+
+
+
+But on page load (e.g. if we refresh the page) the list of tasks will not be shown.
+To make that work, we need to make a `GET` request.
+
+```html
+<div id="todo-list" hx-get="/taskList" hx-trigger="load"></div>
+```
+
+In the above html, I have added `hx-get` with value `/tasklist` and `hx-trigger` with value 'load'. The `load` value specifies when the request is sent in this case it's when the html file is loaded.
+
+```go
+func listTask(w http.ResponseWriter, r *http.Request) {
+    tastList := getTODO() // gets data from database and returns html
+    io.WriteString(w, tastList)
+}
+func main() {
+    ...
+    http.HandleFunc("/save", listTask)
+    ...
+}
+```
+![demo](src/assets/htmx/yMVvdP3.gif)
+
+Now the list is shown even when the page is refreshed.
+
+For edit let's try some thing different. Previously we did all our operations in a single page for edit we will be navigating to another page without refreshing the wholepage unlike normal html page navigation using an anchor tag.
+
+First let's make clicking on the todo item go to a different page that will contain form for updating the todo item.
+For that, we need to get an anchor tag instead of a div with the todo text inside a div
+
+
+
+In the server i will get the task and return the proper form for updating that task.
+
+```go
+func main() {
+    ...
+    http.HandleFunc("/task/", indivisualTask)
+    ...
+}
+```
+
+**example response from server**
+
+```htmx
+  <div hx-boost="true">
+      <a href="/task/asdasda">asdasda</a>
+  </div>
+```
+
+Here the `hx-boost` attribute will replace the body element and add the previous page to the history entry which will allow you to go back if needed.
+
+![demo](src/assets/htmx/bktyFpm.gif)
+
+For the actual editing, it will be similar to the other examples.
+
+```html
+<div>
+  <div>clean room</div>
+  <form id="myForm2" hx-put="/edit" hx-target="#app">
+    <input type="text" name="task" required="" />
+    <input type="hidden" name="old" value="clean room" />
+    <input type="submit" value="edit" />
+  </form>
+</div>
+```
+
+Now let's perform deletion on the list.
+
+First, we have to make a delete button.
+
+For that, I will update the list template and add a delete form
+
+```html
+<div hx-boost="true" class="new-element">
+  <a href="/task/dasdas">dasdas</a>
+  <form
+    hx-trigger="click"
+    hx-delete="/delete"
+    hx-target="closest .new-element"
+    hx-swap="outerHTML swap:1s"
+  >
+    <input type="hidden" name="task" value="dasdas" />
+    delete
+  </form>
+</div>
+```
+
+Most of the snippets in the delete form are similar to the other examples. but in the `hx-swap="outerHTML swap:1s"` attribute I added an extra value . All that value does is delay the swapping of the element by 1 second because i want to do a css animation during that time.
```

Comment:
```suggestion
Most of the snippets in the delete form are similar to the other examples. But in the `hx-swap="outerHTML swap:1s"` attribute I added an extra value. This value is delay the swapping of the element by 1 second because I want to do a CSS animation during that time.
```